scene/main/viewport.cpp

```diff
@@ -3624,18 +3628,40 @@ void SubViewport::set_use_arvr(bool p_use_arvr) {
 
 	RS::get_singleton()->viewport_set_use_arvr(get_viewport_rid(), arvr);
 }
-
 bool SubViewport::is_using_arvr() {
 	return arvr;
 }
 
 void SubViewport::set_size(const Size2i &p_size) {
-	_set_size(p_size, Size2i(), Rect2i(), Transform2D(), true);
+	_set_size(p_size, _get_size_override(), Rect2i(), _stretch_transform(), true);
 }
 Size2i SubViewport::get_size() const {
 	return _get_size();
 }
 
+void SubViewport::set_size_override(const Size2i &p_size) {
```

Comment:
I would rename to:

set_2d_size_override
set_2d_size_override_stretch

which I believe is closer to what it actually does.

Reply:
@reduz Doesn't this also affect 3D rendering (i.e. lower sizes will make it pixelated)?

Reply:
@Calinou Nope, it only affects 2D. I'm going to start making the changes.